nitro-protocol/contracts/ForceMove.sol

```diff
@@ -40,16 +40,18 @@ contract ForceMove is IForceMove, StatusManager {
      * @notice Registers a challenge against a state channel. A challenge will either prompt another participant into clearing the challenge (via one of the other methods), or cause the channel to finalize at a specific time.
      * @dev Registers a challenge against a state channel. A challenge will either prompt another participant into clearing the challenge (via one of the other methods), or cause the channel to finalize at a specific time.
      * @param fixedPart Data describing properties of the state channel that do not change with state updates.
-     * @param signedVariableParts An ordered array of structs, that can be signed by any number of participants, each struct describing the properties of the state channel that may change with each state update.
+     * @param proof An ordered array of structs, that can be signed by any number of participants, each struct describing the properties of the state channel that may change with each state update. The proof is a validation for the supplied candidate.
+     * @param candidate A struct, that can be signed by any number of participants, describing the properties of the state channel to change to. The candidate state is supported by proof states.
      * @param challengerSig The signature of a participant on the keccak256 of the abi.encode of (supportedStateHash, 'forceMove').
      */
     function challenge(
         FixedPart memory fixedPart,
-        SignedVariablePart[] memory signedVariableParts,
+        SignedVariablePart[] memory proof,
+        SignedVariablePart memory candidate,
         Signature memory challengerSig
     ) external override {
         bytes32 channelId = NitroUtils.getChannelId(fixedPart);
-        uint48 largestTurnNum = _lastVariablePart(signedVariableParts).turnNum;
+        uint48 largestTurnNum = candidate.variablePart.turnNum;
```

Comment:
Can we be _sure_ that this turn number is "largest"? Let's just use `candidateTurnNum` or something, and let L56-63 handle turn number comparison logic?

Reply:
Previously, in Nitro V1, we had `largestTurnNum` supplied and it was used to compose the state, treating the last variable part supplied as the one having the `largestTurnNum`.

Now, in V2, we _may_ imply that `candidate` is the state with `largestTurnNum`, thus resembling V1 logic.

On the other hand, that would make it impossible to create DApps with some peculiar logic expecting `candidate` not to have `largestTurnNum`, but that seems weird.

If participants specify the `candidate`, they expect State Channel state to "upgrade" specifiacally to it.

Moreover, if we change the API to expect a separate `largestTurnNum`, how would its check look like? Will we iterate through all supplied `variableParts`, checking if any has `largestVariablePart` in it?

Reply:
I think since we are handing off almost 100% of the responsibility about which state is supported to the application, we should not give this variable a name like "largestTurnNum` **unless** we have some mechanism in the core protocol to ensure that this is an appropriate name. I don't think we should put that mechanism in. Most applications (perhaps all) will do the requisite checks on the turn number (see e.g. `StrictTurnTaking`). But "most likely" is not a strong enough guarantee to use this variable name. 

I don't think we need to add a separate `largestTurnNum` to the API. I propose that we allow for (but discourage) the "weird" logic where an application can allow a later state to form the support proof for an earlier state. 

In Nitro v2 the application and the core protocol must be audited together by a suspicious user. So such weird logic would be rejected/avoided in most cases.

Reply:
Renaming `largestTurnNum` to `candidateTurnNum` then.